File: handler.js
'use strict';

const AWS = require('aws-sdk');
const stepfunctions = new AWS.StepFunctions();

module.exports.executeStepFunction = (event, context, callback) => {
  console.log('executeStepFunction');

  const numberOne = event.queryStringParameters.numberOne;
  const numberTwo = event.queryStringParameters.numberTwo;
  console.log(numberOne);
  console.log(numberTwo);

  callStepFunction(numberOne, numberTwo).then(result => {
    let message = 'Step function is executing';
    if (!result) {
      message = 'Step function is not executing';
    }

    const response = {
      statusCode: 200,
      body: JSON.stringify({ message })
    };

    callback(null, response);
  });
};


function callStepFunction(numberOne, numberTwo) {
  console.log('callStepFunction');

  const stateMachineName = 'SequenceActivityMachine'; // The name of the step function we defined in the serverless.yml
  console.log('Fetching the list of available workflows');

  return stepfunctions
    .listStateMachines({})
    .promise()
    .then(listStateMachines => {
      console.log('Searching for the step function', listStateMachines);

      for (var i = 0; i < listStateMachines.stateMachines.length; i++) {
        const item = listStateMachines.stateMachines[i];

        if (item.name.indexOf(stateMachineName) >= 0) {
          console.log('Found the step function', item);

          var params = {
            stateMachineArn: item.stateMachineArn,
            input: JSON.stringify({ numberOne: numberOne, numberTwo: numberTwo })
          };

          console.log('Start execution');
          return stepfunctions.startExecution(params).promise().then(() => {
            return true;
          });
        }
      }
    })
    .catch(error => {
      return false;
    });
}

File: handler.py
from __future__ import print_function

import os
import json
import urllib
import requests
import boto3
from botocore.exceptions import ClientError

    
def executeStepFunction(event, context):

    ### Launch three Pods ###

    #url = 'http://127.0.0.1:8080/api/v1/namespaces/default/pods'
    #payloadCalTwoNum = open("calculatetwonumber.json")
    #payloadDyFirstMsg = open("displayfirstmessage.json")
    #payloadDySecondMsg = open("displaysecondmessage.json")
    #headers = {'content-type': 'application/json', 'Accept': 'application/json'}

    #requests.post(url, data=payloadCalTwoNum, headers=headers)
    #requests.post(url, data=payloadDyFirstMsg, headers=headers)
    #requests.post(url, data=payloadDySecondMsg, headers=headers)


    ### Get all events from K8S ###

    #url = 'http://127.0.0.1:8080/api/v1/events'
    #response = requests.post(url, headers=headers)


    ### Call State Machines ###
    numberOne = event['queryStringParameters']['numberOne']
    numberTwo = event['queryStringParameters']['numberTwo']
    
    stateMachineName = 'SequenceNumActivityMachine'
    stepFunction = boto3.client('stepfunctions')
    state_machines = stepFunction.list_state_machines()
    message = 'Step function not is executing'
    
    try:
        if state_machines.get('stateMachines'):
            for sm in state_machines.get('stateMachines'):
                if sm['name'] == stateMachineName:
                    stepFunction.start_execution(stateMachineArn=sm['stateMachineArn'], input=json.dumps({ numberOne: numberOne, numberTwo: numberTwo }))
                    message = 'Step function is executing'
            
        body = {
            "message": message,
            "input": event
        }

        response = {
            "statusCode": 200,
            "body": json.dumps(body)
        }
    except ClientError as e:
        body = {
            "message": e.response['Error']['Code'],
            "input": event
        }

        response = {
            "statusCode": 400,
            "body": json.dumps(body)
        }

    return response
    

    # Use this code if you don't use the http event with the LAMBDA-PROXY
    # integration
    """
    return {
        "message": "Go Serverless v1.0! Your function executed successfully!",
        "event": event
    }
    """

def ExecuteErrorInExecution(event, context):

    logger.info("Starting ExecuteErrorInExecution ...")
    logger.info(event)

    # activity = "arn:aws:states:us-east-2:912778368198:activity:calculateTwoNumber"
    # stepFunction = boto3.client('stepfunctions')
    # task = stepFunction.get_activity_task(activityArn=activity, workerName="CalculateTwoNumberNext")
    # taskToken=task['taskToken']

    # response = stepFunction.send_task_failure(
    #     taskToken=taskToken,
    #     error='task token fail ...',
    #     cause='dont know the reason'
    # )

    err_resp = {
        "statusCode": 'event error',
        "body": 'event cause'
    }
    return err_resp
